realm/src/main/java/io/realm/RealmConfiguration.java

```diff
@@ -0,0 +1,390 @@
+/*
+ * Copyright 2015 Realm Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.realm;
+
+import android.content.Context;
+
+import java.io.File;
+import java.lang.reflect.Constructor;
+import java.lang.reflect.InvocationTargetException;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.Set;
+
+import io.realm.annotations.RealmModule;
+import io.realm.exceptions.RealmException;
+import io.realm.internal.RealmProxyMediator;
+import io.realm.internal.modules.CompositeMediator;
+import io.realm.internal.modules.FilterableMediator;
+
+/**
+ * A RealmConfiguration is used to setup a specific Realm instance.
+ *
+ * Instances of a RealmConfiguration can only created by using the {@link io.realm.RealmConfiguration.Builder} and calling
+ * its {@link io.realm.RealmConfiguration.Builder#build()} method.
+ *
+ * A commonly used RealmConfiguration can easily be accessed by first saving it as
+ * {@link Realm#setDefaultConfiguration(RealmConfiguration)} and then using {@link io.realm.Realm#getDefaultInstance()}.
+ *
+ * A minimal configuration can be created using:
+ *
+ * {@code RealmConfiguration config = new RealmConfiguration.Builder(getContext()).build())}
+ *
+ * This will create a RealmConfiguration with the following properties
+ * - Realm file is called "default.realm"
+ * - It is saved in Context.getFilesDir()
+ * - It has its schema version set to 0.
+ */
+public class RealmConfiguration {
+
+    private static final Object DEFAULT_MODULE;
+    private static final RealmProxyMediator DEFAULT_MODULE_MEDIATOR;
+    static {
+        DEFAULT_MODULE = getDefaultModule();
+        if (DEFAULT_MODULE != null) {
+            DEFAULT_MODULE_MEDIATOR = getModuleMediator(DEFAULT_MODULE.getClass().getCanonicalName());
+        } else {
+            DEFAULT_MODULE_MEDIATOR = null;
+        }
+    }
+
+    private final File realmFolder;
+    private final String realmFileName;
+    private final String canonicalPath;
+    private final byte[] key;
```

Comment:
Not really. Core does the same thing. It is an attack vector, but the current policy is that we only want to guard against attacks on the filesystem.